src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/UpdateVCFSequenceDictionary.java

```diff
@@ -167,27 +167,43 @@ public void closeTool() {
         }
     }
 
-    // either throws or returns a valid seq dict
-    private SAMSequenceDictionary getSequenceDictionaryFromInput(final String source) {
-        SAMSequenceDictionary dictionary;
-        if (source == null) {
-            if (hasReference()) {
-                dictionary = getReferenceDictionary();
+    // Override getBestAvailableSequenceDictionary() to ensure that the new sequence dictionary specified by
+    // the user (and not the sequence dictionary from the VCF we're trying to update) is used uniformly by all
+    // callers. Otherwise, the wrong dictionary would be used when writing the index for the output vcf.
+    //
+    @Override
+    public SAMSequenceDictionary getBestAvailableSequenceDictionary() {
+
+        SAMSequenceDictionary resultDictionary;
+
+        final SAMSequenceDictionary masterDictionary = getMasterSequenceDictionary();
+        if (dictionarySource == null) {
+            if (masterDictionary != null) {
+                // We'll accept the master dictionary if one was specified. Using the master dictionary
```

Comment:
Can you throw in a logging statement that the dictionary is being overridden?  I know it'll be captured in the arguments used to call a tool, but it seems like another explicit warning would be good (particularly in cases where the specified dictionary is different from the embedded dictionary in the file / etc).

Reply:
Yeah, done.